### PR TITLE
vimc-4929: Ensure workflow execution isolates failure

### PR DESCRIPTION
**Pull request by r-ash**
- [x] spec.md has been updated or doesn't need to updated

This PR will
* Add a test to confirm that if we run a workflow which contains some independent and some dependent reports if 1 report errors it only cancels run of dependent reports and not independent reports

This will fail until we merge https://github.com/mrc-ide/rrq/pull/62

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -17,11 +17,9 @@ Imports:
     porcelain (>= 0.1.4),
     processx,
     redux,
-    rrq (>= 0.5.5),
+    rrq (>= 0.5.6),
     utils,
-    webutils,
```
Comment (richfitz) [inline DESCRIPTION:22]: What happened here? Why do we drop webutils and zip?

File: inst/schema/spec.md [hunk 2]
```diff
@@ -484,7 +484,7 @@ Response
 
 ## POST /workflow/run
 
-Run a set of reports as a workflow. Takes a set of reports to run, and any instance, params or dependencies between the reports. Works out the order the reports can be sent to the queue respecting any dependencies and adds them to the queue. Returns the ID of the workflow and each individual report.
+Run a set of reports as a workflow. Takes a set of reports to run, and any instance, params or dependencies between the reports. Works out the order the reports can be sent to the queue respecting any dependencies and adds them to the queue. Returns the ID of the workflow and each individual report. If 1 report fails reports dependent on the failed report will have status set to "impossible". Any independent reports will still be run.
```
Comment (richfitz) [inline inst/schema/spec.md:487]: Should this be controllable by a parameter? Seems like something that we'd not always want?

File: R/runner.R [hunk 3]
```diff
@@ -78,6 +78,17 @@ runner_run <- function(key_report_id, key, root, name, parameters, instance,
     Sys.sleep(poll)
   }
 
+  ## For very fast report running we can get into the state that
```
Comment (richfitz) [inline R/runner.R:81]: I think this belongs within the above loop already? So an `if (is.na(id)) next` or similar at ~74

**Review by richfitz — CHANGES_REQUESTED** [review]

File: DESCRIPTION (hunk 1 above)
Comment (r-ash) [reply DESCRIPTION:22]: This was just cleaning up an R CMD CHECK note, webutils no longer used and zip only used in tests

File: inst/schema/spec.md (hunk 2 above)
Comment (r-ash) [reply inst/schema/spec.md:487]: Yeah might be useful, though I've not heard of a use case from anyone where they want that behaviour. I was thinking could alternatively add a "cancel" button so a running workflow can be stopped.

**Review by richfitz — APPROVED** [review]

**Pull request merged**